internal/auth0/resourceserver/resource_scopes.go

```diff
@@ -0,0 +1,228 @@
+package resourceserver
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/auth0/terraform-provider-auth0/internal/config"
+	"github.com/auth0/terraform-provider-auth0/internal/value"
+)
+
+// NewScopesResource will return a new auth0_resource_server_scopes (1:many) resource.
+func NewScopesResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"resource_server_identifier": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Identifier of the resource server that the scopes (permission) are associated with.",
+			},
+			"scopes": {
```

Comment:
I know we've perpetuated this convention but wondering if it makes sense to have this be `scope` (singular). For example:

```resource "auth0_resource_server_scopes" "my_api_scopes" {
  resource_server_identifier = auth0_resource_server.my_api.identifier

  scope {
    name        = "create:appointments"
    description = "Ability to create appointments"
  }

  scope {
    name        = "read:appointments"
    description = "Ability to read appointments"
  }
}
```

IMO it reads better to define each individual scope than defining individual scopes. But wether its better to be consistent or more standard to Terraform conventions I'm not sure.